File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    q = {numbers[0]:1, -numbers[0]:1}
    for i in range(1, len(numbers)):
        dct = {}
        for idx, cnt in q.items():
            if dct.get(idx+numbers[i]):
                dct[idx+numbers[i]] += cnt
            else:
                dct[idx+numbers[i]] = cnt
            if dct.get(idx-numbers[i]):
                dct[idx-numbers[i]] += cnt
            else:
                dct[idx-numbers[i]] = cnt
        q = dct
        dct = {}
    if q.get(target):
        return q[target]
    else:
        return 0

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int total;

void DFS(int n, int sum, int &target, vector<int> &v){
    if (n >= v.size()) {
        if (sum == target) {
            total++;
        }
    } else {
        DFS(n+1, sum+v[n], target, v);
        DFS(n+1, sum-v[n], target, v);
    }
}

int solution(vector<int> numbers, int target) {
    DFS(0, 0, target, numbers);
    return total;
}